docs/react/guides/nextjs.md

```diff
@@ -142,3 +142,75 @@ export default Index;
 
 You can share React components between multiple Next.js applications. You can also share web components between Next.js and plain React applications. You can even share code between the backend and the frontend. All can be done without any unnecessary ceremony.
+
+## Deploying to Vercel
+
+You may know that the company behind Next.js, Vercel, has a great hosting platform offering that is developed in tandem with Next.js itself to offer a great overall developer and user experience.
+
+In order to deploy your Next.js application from your Nx workspace you should do the following:
+
+### Verify the project's next.config.js
+
+Let's continue to use our `tuskdesk` example from above, and so we need to check out our config at `apps/tuskdesk/next.config.js`. If you created the application using a recent (at the time of writing) version of Nx, such as Nx 11, then you will likely see the following in that config by default:
+
+```js
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const withNx = require('@nrwl/next/plugins/with-nx');
+
+module.exports = withNx({});
+```
+
+If you have a config which looks like that, leveraging the `withNx()` config plugin, **no further action is needed** in your config.
+
+If, however, you created the application using an older version of Nx, you may just see an empty object:
+
+```js
+module.exports = {};
+```
+
+If this is the case, then you must do one of the following:
+
+[Option 1] We would naturally highly recommend upgrading to the latest Nx (for many reasons), and updating the next.config.js to match the first example which leverages the `withNx()` config plugin.
+
+[Option 2] If for some reason you cannot upgrade to a version of Nx which provides the `withNx()` config plugin, you can manually add a `target` property to your exported config with a value of `'experimental-serverless-trace'`.
+
+E.g.
+
+```js
+module.exports = {
+  target: 'experimental-serverless-trace',
+  // ...You can of course have other Next.js config options specified here too, but the "target" is critical for Vercel deployments...
+};
+```
+
+> Vercel themselves have informed us that this target will not be required in future versions of Next.js and their platform, but that even when that is the case this option will not cause any issues, so we do not need to worry too much about the name containing "experimental".
+
+### Configure your Vercel project's settings appropriately
+
+#### New Vercel project
+
+1. If you are "importing" your Nx workspace's repository for the first time, make sure you do _not_ choose a root directory as part of the repo selection process (therefore leaving it to be the root of the full repo/workspace)
+2. Ensure the Next.js "Framework Preset" is selected
+3. Expand the "Build and Output Settings" and toggle the override switch for the build command. For our `tuskdesk` project the value will look like this:
+
+```sh
+npx nx build tuskdesk --prod --outputPath=.
```

Comment:
I'll let the Nrwl folks decide on what they want to do about this but if this was the case for you then it means you must have changed your setup to remove the production configuration that was generated for you, so IMO it seems correct that you also have to perform custom actions when following official guides...

Reply:
We should probably leave `--prod` in the docs since it is more consistent with the rest of the Nx tooling, and it works with newly generated apps.

Some folks may use the `fileReplacements` option to replace files for different configurations.